tests/python/pants_test/backend/jvm/tasks/jvm_compile/test_jvm_classpath_published.py

```diff
@@ -79,3 +98,85 @@ def test_incremental_caching(self):
             os.pathsep.join([os.path.join(jar_dir, 'z2.jar'), os.path.join(jar_dir, 'z3.jar')]) + '\n'
           ]
         )
+
+  def _assert_internal_classpath_option(self, *, manifest_jar: bool, internal_classpath_only: bool) -> None:
+    with temporary_dir(root_dir=self.pants_workdir) as jar_dir, \
+         temporary_dir(root_dir=self.pants_workdir) as dist_dir:
+      self.set_options(pants_distdir=dist_dir,
+                       internal_classpath_only=internal_classpath_only,
+                       manifest_jar=manifest_jar)
+
+      self.add_to_build_file('java/classpath', dedent("""\
+      jar_library(
+        name='jar-lib',
+        jars=[
+          jar(org='commons-io', name='commons-io', rev='2.6'),
+        ],
+      )
+
+      java_library(
+        name='java-lib',
+        sources=['com/foo/Bar.java'],
+        dependencies=[
+          ':jar-lib',
+        ],
+      )
+      """))
+      jar_lib = self.target('java/classpath:jar-lib')
+      init_target = self.target('java/classpath:java-lib')
+      context = self.context(target_roots=[jar_lib, init_target])
+      runtime_classpath = context.products.get_data('runtime_classpath',
+                                                    init_func=ClasspathProducts.init_func(self.pants_workdir))
+      task = self.create_task(context)
+
+      target_classpath_output = os.path.join(dist_dir, self.options_scope)
+
+      # Create a classpath entry.
+      touch(os.path.join(jar_dir, 'jar-lib-target.jar'))
+      touch(os.path.join(jar_dir, 'java-lib-target.jar'))
+
+      cache_path = os.path.join(jar_dir, 'jar-lib-target.jar')
+      runtime_classpath.add_for_target(jar_lib, [(self.DEFAULT_CONF, ArtifactClasspathEntry(
+        path=cache_path,
+        coordinate=M2Coordinate(org='commons-io', name='commons-io', rev='2.6'),
+        cache_path=cache_path,
+      ))])
+      runtime_classpath.add_for_target(init_target, [(self.DEFAULT_CONF, os.path.join(jar_dir, 'java-lib-target.jar'))])
+
+      task.execute()
+
+      if manifest_jar:
+        if internal_classpath_only:
+          expected_num_artifacts = 1
+        else:
+          expected_num_artifacts = 2
+        manifest_jar_path = os.path.join(target_classpath_output, 'manifest.jar')
+        with open_zip(manifest_jar_path) as zf:
+          with zf.open('META-INF/MANIFEST.MF') as manifest_file:
+            all_output = ''.join(
+              re.sub(r'^(Class-Path:)? |\n+', '', line.decode('utf-8')) for line in manifest_file
+            ).split(' ')
```

Comment:
Not a blocker: could possibly consider using a library here to parse the manifest file.